Makefile

```diff
@@ -62,6 +62,11 @@ ifdef CUDA_FLAGS
 CMAKE_FLAGS += -DCMAKE_CUDA_FLAGS="$(CUDA_FLAGS)"
 endif
 
+# Prefer using clang, if available, over default gcc
+ifneq (, $(shell which clang))
```

Comment:
We should not override user preferences. I am not too sure if we should add this block.

Reply:
Any basic install will have gcc installed. Not sure how we make sure it is also used other than hard override the cmake directly (and there was something about a difference doing it in the CMakeLists.txt or on the command line).

Either way, you suggest new targets in the `Makefile` instead of using the override? Something like
```
clang-relase
clang-debug
```

Reply:
I think we should test Clang17 performance (both code and build) before we make it the default.
Until then users or CI can set `CXX` in the environment or use `CMAKE_CXX_COMPILER` if they want to use clang17.

Reply:
Would having the new targets be a bit more user friendly for now?

Reply:
Yes, adding makefile targets should help.

Reply:
Please see the latest changes.